services/galley/src/Galley/API/Teams.hs

```diff
@@ -336,6 +337,29 @@ updateTeamH zusr zcon tid updateData = do
   let r = list1 (userRecipient zusr) (membersToRecipients (Just zusr) (memList ^. teamMembers))
   E.push1 $ newPushLocal1 (memList ^. teamMemberListType) zusr (TeamEvent e) r & pushConn ?~ zcon
 
+updateTeamDeleteSplashScreenH ::
+  Members
+    '[ ErrorS 'NotATeamMember,
+       ErrorS ('MissingPermission ('Just 'SetTeamData)),
+       GundeckAccess,
+       Input UTCTime,
+       TeamStore
+     ]
+    r =>
+  UserId ->
+  ConnId ->
+  TeamId ->
+  Sem r ()
+updateTeamDeleteSplashScreenH zusr zcon tid = do
+  zusrMembership <- E.getTeamMember tid zusr
+  void $ permissionCheckS SSetTeamData zusrMembership
+  E.deleteTeamSplashScreen tid
```

Comment:
I'm not convinced this is the right approach, but the alternative of changing `TeamUpdateData` to accomodate both leaving the splash screen intact (`Nothing`) and setting it to `Nothing` (???) doesn't appeal to me either.  We could change the `Maybe` to something like:

```haskell
data ChangeSplashScreen = Keep | Delete | Set AssetKey
```

But then how would that translate into json?  like this?:

```
{ ... 
  "splash_screen": null | {"delete"} | {"set": "key"},
  ...
}
```

Reply:
I mean:

```
  "splash_screen": null | {"delete": true} | {"set": "key"},
```

Reply:
I ended up doing what Patryk suggested earlier, namely do the same thing we're already doing for the Icon (aka Logo) [here](https://github.com/wireapp/wire-server/blob/5b0a276051990c4ff51eeff77798ab15715cd059/libs/wire-api/src/Wire/API/Team.hs#L247).